File: WePeiYang/lostfound/src/main/java/com/twtstudio/tjwhm/lostfound/mylist/MylistPagerAdapter.java
package com.twtstudio.tjwhm.lostfound.mylist;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by tjwhm on 2017/7/4.
 **/

public class MylistPagerAdapter extends FragmentPagerAdapter {

    List<Fragment> fragmentsOfMylist = new ArrayList<>();
    List<String> fragmentsTitles = new ArrayList<>();
    public MylistPagerAdapter(FragmentManager fm) {
        super(fm);
    }
    public void add(Fragment fragment,String title){
        fragmentsOfMylist.add(fragment);
        fragmentsTitles.add(title);
    }

    @Override
    public Fragment getItem(int position) {
        return fragmentsOfMylist.get(position);
    }

    @Override
    public int getCount() {
        return fragmentsOfMylist.size();
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return fragmentsTitles.get(position);
    }
}

File: WePeiYang/lostfound2/src/main/java/com/yookiely/lostfond2/mylist/MyListPagerAdapter.kt
package com.yookiely.lostfond2.mylist

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter
import java.util.ArrayList

class MyListPagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {

    private var fragmentsOfMyList: MutableList<Fragment> = ArrayList()
    private var fragmentsTitles: MutableList<String> = ArrayList()

    fun add(fragment: Fragment, title: String) {
        fragmentsOfMyList.add(fragment)
        fragmentsTitles.add(title)
    }

    override fun getItem(position: Int): Fragment = fragmentsOfMyList[position]

    override fun getCount(): Int = fragmentsOfMyList.size

    override fun getPageTitle(position: Int): CharSequence = fragmentsTitles[position]
}

